### PR TITLE
Add Basic Buildcraft support

### PR DESCRIPTION
**Pull request by fionera**
This PR adds support for all interactions that can be done with Gates in Buildcraft. Sadly not all Blocks implement the needed Interfaces or I just didnt saw it.

### CODE REVIEW

File: src/main/java/org/squiddev/plethora/integration/buildcraft/TileControllable.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	@PlethoraMethod(modId = BCCore.MODID, doc = "-- Get the control mode.")
+	public static String getControlMode(@FromTarget IControllable controllable) {
+		return controllable.getControlMode().lowerCaseName;
+	}
+
+	@PlethoraMethod(modId = BCCore.MODID, doc = "-- Reports true if the Block accept the given mode.")
+	public static boolean acceptsControlMode(@FromTarget IControllable controllable, String modeName) throws LuaException {
+		IControllable.Mode mode;
+		try {
```
Comment (SquidDev) [inline src/main/java/org/squiddev/plethora/integration/buildcraft/TileControllable.java]: I _think_ you can use `IControllable.Mode` directly on the argument. We definitely do it for other enums:

https://github.com/SquidDev-CC/plethora/blob/527d945ab4c69514a63d2a04e2fcb767b8cfb3f8/src/main/java/org/squiddev/plethora/integration/rf/MethodsEnergy.java#L30

File: src/main/java/org/squiddev/plethora/integration/buildcraft/TileControllable.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import dan200.computercraft.api.lua.LuaException;
+import org.squiddev.plethora.api.Injects;
+import org.squiddev.plethora.api.method.wrapper.FromTarget;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Injects(BCCore.MODID)
```
Comment (SquidDev) [inline src/main/java/org/squiddev/plethora/integration/buildcraft/TileControllable.java]: You don't need `@Injects` if you've only got methods inside the class. `@Injects` is used for fields.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Thank you very much for the PR!

Just left a couple of comments on places where annotation magic can be used or isn't needed, otherwise looks really good.

**Comment by fionera** [comment]
I did both changes but cant test them currently because my OpenGL is suddently broken. Will push when its tested ^^

**Review by SquidDev — APPROVED** [review]

**Comment by SquidDev** [comment]
Thank you very much!

**Comment by fionera** [comment]
That was fast :) Thanks for merging. Gonna backport it now to support the older CC:T Version I use ^^

**Pull request merged**